test/langtools/tools/lib/toolbox/ToolBox.java

```diff
@@ -264,37 +264,32 @@ public void copyDir(String fromDir, String toDir) {
 
     /**
      * Copies the contents of a directory to another directory.
+     * The destination direction should not already exist.
      * <p>Similar to the shell command: {@code rsync fromDir/ toDir/}.
      *
      * @param fromDir the directory containing the files to be copied
      * @param toDir   the destination to which to copy the files
      */
     public void copyDir(Path fromDir, Path toDir) {
         try {
-            Files.createDirectories(toDir);
-            // See second example in java.nio.file.FileVisitor, Usage Examples
-            Files.walkFileTree(fromDir, EnumSet.of(FileVisitOption.FOLLOW_LINKS), Integer.MAX_VALUE,
-                    new SimpleFileVisitor<Path>() {
-                        @Override
-                        public FileVisitResult preVisitDirectory(Path fromSubdir, BasicFileAttributes attrs)
-                                throws IOException {
-                            Path toSubdir = toDir.resolve(fromDir.relativize(fromSubdir));
-                            try {
-                                Files.copy(fromSubdir, toSubdir);
-                            } catch (FileAlreadyExistsException e) {
-                                if (!Files.isDirectory(toSubdir))
-                                    throw e;
-                            }
-                            return FileVisitResult.CONTINUE;
-                        }
-
-                        @Override
-                        public FileVisitResult visitFile(Path fromFile, BasicFileAttributes attrs)
-                                throws IOException {
-                            Files.copy(fromFile, toDir.resolve(fromDir.relativize(fromFile)));
-                            return FileVisitResult.CONTINUE;
-                        }
-                    });
+            if (toDir.getParent() != null) {
+                Files.createDirectories(toDir.getParent());
+            }
```

Comment:
This is exactly what I was going to re-suggest after I read your reply on my initial suggestion. We're on the same page. Thanks for simplifying this stuff.

Reply:
Actually, we could simplify it to an unconditional  `Files.createDirectories(toDir.toAbsolutePath().getParent())`

Reply:
Hm...
```
jshell> Path.of("/").getParent()
$1 ==> null

jshell> Path.of("/").toAbsolutePath().getParent()
$2 ==> null

jshell> /reset
|  Resetting state.

jshell> Path.of("/").getParent()
$1 ==> null

jshell> Path.of("/").toAbsolutePath().getParent()
$2 ==> null

jshell> Files.createDirectories(null)
|  Exception java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileSystem()" because "path" is null
|        at Files.provider (Files.java:105)
|        at Files.createDirectory (Files.java:700)
|        at Files.createAndCheckIsDirectory (Files.java:807)
|        at Files.createDirectories (Files.java:753)
|        at (#3:1)
```